pkg/arvo/sys/arvo.hoon

```diff
@@ -693,26 +694,49 @@
               =^  rey  +>+  (^load +<)
               [rey +>.$]
     ::
-    ++  peek  |=(* (^peek ;;([@da path] +<)))      ::  46
+    ++  peek  |=  *                                     ::  46
+              =/  rob  (^peek ;;([@da path] +<))
+              ?~  rob  ~
+              ?~  u.rob  ~
+              [~ u.u.rob]
     ::
     ++  poke  |=  *                                     ::  47
-              ^-  [(list ovum) *]
               =>  .(+< ;;([now=@da ovo=ovum] +<))
-              =^  ova  +>+.$  (^poke now ovo)
+              =;  taq=_^?(|.(**))
+                =/  res
+                  %+  mock
+                    [taq 9 2 0 1]
+                  |=  [ref=* raw=*]
+                  (^peek now ;;(path raw))
+                |-
+                ?-  -.res
+                    %0  p.res
+                    %1  $(res [%2 leaf+"blocked" >((list path) p.res)< ~])
+                    %2
+                  ~_  leaf+"deterministic error"
+                  |-
+                  ?~  p.res
+                    !!
+                  ~_  i.p.res
+                  $(p.res t.p.res)
+                ==
```

Comment:
@philipcmonk, on second thought, aren't these changes now superfluous? %gall virtualizes app invocations and sets the scry handler. This approach works to a) pin `.^` for the duration of an event and b) enables `.^` in vanes. The former is desirable, but only a partial solution to a pinned scry, while the latter is more debatable. It seems like this should be reverted.

Reply:
You're correct, I'll remove it.